geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/ExecuteFunction.java

```diff
@@ -55,6 +58,25 @@ public static Command getCommand() {
     return singleton;
   }
 
+  private final InternalFunctionExecutionService internalFunctionExecutionService;
+  private final ServerToClientFunctionResultSenderFactory serverToClientFunctionResultSenderFactory;
+  private final FunctionContextImplFactory functionContextImplFactory;
+
+  private ExecuteFunction() {
+    this(InternalFunctionService.getInternalFunctionExecutionService(),
+        new DefaultServerToClientFunctionResultSenderFactory(),
+        new DefaultFunctionContextImplFactory());
+  }
+
+  @TestingOnly
```

Comment:
Why not just use this constructor in the production code as well, and inject the dependencies there?  Do we need two constructors (one of which is a test hook)?